docs/serving/using-auto-tls.md

```diff
@@ -6,10 +6,35 @@ type: "docs"
 ---
 
 If you install and configure cert-manager, you can configure Knative to
-automatically obtain new TLS certificates and renew existing ones. To learn more
-about using secure connections in Knative, see
+automatically obtain new TLS certificates and renew existing ones for Knative 
+Service.
+To learn more about using secure connections in Knative, see
 [Configuring HTTPS with TLS certificates](./using-a-tls-cert.md).
 
+## Automatic TLS provision mode
+
+In Knative, we support the following modes of Auto TLS:
+
+1.  Using DNS01 challenge
+
+    In this mode, your cluster needs to be able to talk to your DNS server to verify the ownership of your domain.
+    Specifically, when using DNS challenge, we support:
+    - **Provision Certificate per namespace:**
+      - In this mode, a single Certificate will be provisioned per namespace and be reused across the Knative 
+      Services within the same namespace if possible.
+      - If you want to have a fast certificate provision process, this way is 
+      recommended.
+
+    - **Provision Certificate per Knative Service:**
+      - In this mode, a Certificate will be provisioned for each Knative Service.
+      - If you want to have better certificate isolation between each Knative Service, you can choose this way.
+      - The TLS effective time is longer as it needs Certificate provision for each Knative Service creation.
+
+1.  Using HTTP01 challenge
+
+    - In this type, your cluster does not need to be able to talk to your DNS server. You just need to map your domain to the IP of the cluser ingress.
+    - When using HTTP01 challenge, **Certificate will be provisioned per Knative Service.** Certificate provision per namespace is not supported when using HTTP01 challenge. 
+
 ## Before you begin
 
 You must meet the following prerequisites to enable automatic certificate
```

Comment:
choose one term to refer to "automatic certificate" and stick with it, e.g. either use "Auto-TLS" or something like "automatic certificate provisioning".